lib/tasks/deleted_objects.rake

```diff
@@ -11,4 +11,16 @@ namespace :deleted_objects do
       sleep(0.5) unless Rails.env.test?
     end
   end
+
+  desc 'Destroy all the stale objects whose owner is also a deleted object'
+  task :destroy_the_stale_object => :environment do
+    delete_all_the_stale_object_with_owner_service = DeletedObject.where.has do
+      ((id.in DeletedObject.deleted_owner.select(:id)) | (object_type == Service.name)) \
+      & (created_at <= 1.week.ago)
+    end
+    delete_all_the_stale_object_with_owner_service.find_in_batches(batch_size: 300) do |records|
```

Comment:
```suggestion
    stale_or_service_objects.find_in_batches(batch_size: 300) do |records|
```
Btw, given my experience so far with this table, we can make it a bigger batch size. At least 500 😄